### PR TITLE
Add Microsoft Installer (MSI) package to releases

### PR DESCRIPTION
**Pull request by briantist**
This PR modifies the GitHub workflow for releasing to run an additional job that builds an MSI package, and adds it to the release assets (only for AMD64).

The existing build is already taking care of a Windows binary, and there's chocolatey installs available, but MSI packages offer a number of advantages:
- Standard packing format that natively supports install/remove/repair
- Works interactively or unattended
- Works with many software management platforms in enterprises
- Works with Group Policy

The MSI is built with the [Wix toolset](https://wixtoolset.org/) in Wine and does not require a Windows runner or environment to build, and this uses a pre-built container that is set up with wix.

The existing build and release steps using go-releaser are not changed; the MSI part runs in a separate dependent job, and downloads the .exe from the release produced by the original job, and uploads the resulting .msi as an additional release asset. Here's an example as run on my fork:
- https://github.com/briantist/saml2aws/releases/tag/v2.34.0
- and the full run that produced that: https://github.com/briantist/saml2aws/actions/runs/1792307741

Additionally, I've added a `workflow_dispatch` trigger that takes a `tag` input. When triggered this way, you can run the workflow manually against an existing previous release, to add the MSI build to it after the fact:

image

During that run, it will not re-run the `release` job (it would fail because go-releaser gets errors trying to upload the assets that already exist), and instead will only run the MSI portion. A run like that can be seen here:
https://github.com/briantist/saml2aws/actions/runs/1792301340

And again this will not change the tag of the existing release, it will use the .exe already uploaded to it to produce the MSI.

So if this PR is accepted, that is a way to go back and add MSIs to a few past releases if desired.

---

Happy to answer any questions or concerns, thanks!

### CODE REVIEW

**Comment by briantist** [comment]
@wolfeidau any chance you might be able to take a look this week? Thank you!

**Comment by briantist** [comment]
@wolfeidau sorry to ping directly again, but I haven't seen anyone else reviewing or merging PRs, is there anyone who could do so? Thanks for your work on this project!

**Comment by briantist** [comment]
@versent @lincheney @mapkon is there anything we can do to get some eyes 👀 on this?

**Comment by briantist** [comment]
Hi @wolfeidau ! Noticed some new activity on this repo, rebased my branch. Would really appreciate a review!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -26,3 +32,58 @@ jobs:
...
+        fi
+
+        VERSION=${VER//v}
+
+        echo "VER_TAG=$VER" >> $GITHUB_ENV
+        echo "VERSION=$VERSION" >> $GITHUB_ENV
+        echo "ASSET=saml2aws_${VERSION}_windows_amd64.zip" >> $GITHUB_ENV
+
+    - name: Check out code
+      uses: actions/checkout@v2
```
Comment (briantist) [inline .github/workflows/release.yml:63]:
```suggestion
      uses: actions/checkout@v3
```

**Comment by briantist** [comment]
Hi @wolfeidau @versent checking in again to see if we could get anyone to take a look at this.

There are no code changes in this PR, it doesn't negatively affect the existing releases (even if it breaks), and it'd be a real win for managed Windows environments, admins, and end users.

**Comment by briantist** [comment]
Trying to give this another bump.. @wolfeidau @versent what could I do to get this looked at?

**Comment by briantist** [comment]
@wolfeidau @versent what could I do here to get someone's attention? I'd love to discuss this and open up some kind of conversation at the very least. Please let me know! 🙏

I believe this is a useful change, and I've gone out of my way to make it unobtrusive to any existing process in place. It would help for issues like #820 .

**Comment by briantist** [comment]
would still love to have someone look at this... or chat about it.. @wolfeidau @versent is there anyone who could take a look?

**Comment by briantist** [comment]
Hi @mapkon @wolfeidau @lincheney @versent , I've just updated the PR to keep it up to date with `master`, would anyone take a look at this?

If the intention is is not to consider this PR, it would be nice to know that as well.. thanks

**Comment by mapkon** [comment]
> Hi @mapkon @wolfeidau @lincheney @Versent , I've just updated the PR to keep it up to date with `master`, would anyone take a look at this?
> 
> If the intention is is not to consider this PR, it would be nice to know that as well.. thanks

@briantist Thank you for the message. I will take a look at the earliest. In the long run, we are formulating a much more sustainable maintenance plan for the project, which will involve a contribution path for the wider community. We hope that will hopefully lead to better cadences when it comes to merging and fixing issues.

Please expect a discussion post soon.

**Comment by mapkon** [comment]
@briantist The changes in the #962 will need to be merged with yours

**Comment by briantist** [comment]
> @briantist The changes in the #962 will need to be merged with yours

Thank you, I will take a look next week and incorporate those changes. Much appreciated!

**Comment by briantist** [comment]
@mapkon am I reading #962 correctly or did I miss something? It seems like that PR effectively removed Windows builds by not including them in the matrix.

If that's so, I can look at re-introducing those too.

**Comment by mapkon** [comment]
> @mapkon I've resolved the conflicts and incorporated the changes from #962 .
> 
> I've re-tested on my fork:
> 
> * Here's a "new" release run (note: the annotations about node.js and set-output were resolved in [b1ea919](https://github.com/Versent/saml2aws/commit/b1ea91909bd22c717009db2b751e2c54a27e648d)): https://github.com/briantist/saml2aws/actions/runs/4347276063
> * Here's a run using a workflow dispatch trigger to add the MSI to an existing release: https://github.com/briantist/saml2aws/actions/runs/4347349959
> * Here's the release in question: https://github.com/briantist/saml2aws/releases/tag/v99.0.2
> 
> The `v99.0.2` release did have the MSI in it already from the "new" release run, and then I deleted just the MSI from the assets to simulate an older/existing release, and ran the workflow dispatch against it to show adding an MSI (back) to it.
> 
> Please take a look when you can and let me know if you have any questions or concerns. Thanks!

Thank you! Apologies but I am seeking internal guidance on how to proceed on this given that I am new to this project. I will revert before weeks end

**Comment by briantist** [comment]
> Thank you! Apologies but I am seeking internal guidance on how to proceed on this given that I am new to this project. I will revert before weeks end

@mapkon no problem, thanks for letting me know.
I'm very happy to answer any additional questions or walk through this.

Changes to workflows or the release process can be really tricky to evaluate by PR alone, so if there's additional testing/demonstration I can do on my fork that would be helpful, let me know.

**Comment by mapkon** [comment]
@briantist @gliptak I have moved the discussion here: https://github.com/Versent/saml2aws/discussions/987

Please tag anyone you think can add benefit and thoughts on this. Thank you

**Comment by mapkon** [comment]
Decisions made here: https://github.com/Versent/saml2aws/discussions/987

I will defer this to the next release to stagger the testing.

**Pull request merged**